extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -664,22 +666,23 @@ private static HttpServerOptions createSslOptions(HttpBuildTimeConfig buildTimeC
         return serverOptions;
     }
 
-    private static KeyStoreOptions createKeyStoreOptions(Path keyStorePath, String password, Optional<String> keyStoreFileType,
-            Optional<String> keyStoreProvider, Optional<String> keyStoreAlias) throws IOException {
+    private static KeyStoreOptions createKeyStoreOptions(Path path, String password, Optional<String> fileType,
+            Optional<String> provider, Optional<String> alias, Optional<String> aliasPassword) throws IOException {
         final String type;
-        if (keyStoreFileType.isPresent()) {
-            type = keyStoreFileType.get().toLowerCase();
+        if (fileType.isPresent()) {
+            type = fileType.get().toLowerCase();
         } else {
-            type = findKeystoreFileType(keyStorePath);
+            type = findKeystoreFileType(path);
         }
 
-        byte[] data = getFileContent(keyStorePath);
+        byte[] data = getFileContent(path);
         KeyStoreOptions options = new KeyStoreOptions()
                 .setPassword(password)
                 .setValue(Buffer.buffer(data))
                 .setType(type.toUpperCase())
-                .setProvider(keyStoreProvider.orElse(null))
-                .setAlias(keyStoreAlias.orElse(null));
+                .setProvider(provider.orElse(null))
+                .setAlias(alias.orElse(null))
+                .setAliasPassword(aliasPassword.orElse(null));
```

Comment:
Should we detect when there is an alias password but no alias set?

Reply:
Looking at https://github.com/eclipse-vertx/vert.x/pull/3982/files#diff-97fd41ad7f8a05c475b28e6178697a7eea12daff44699246b33708ab14bbaa4f, the name `aliasPassword` in vert.x might be a bit misleading, because it seems it is used regardless of whether an alias is set or not.
But let me verify...

Reply:
@cescoffier verified: you can use that password without any alias, e.g. image you have a keystore with a single key that has another password than the store. In that case you don't need an alias, but you have to set the key-password.

Therefore, the naming in vert.x (aliasPassword) is a bit misleading but on the Quarkus side of things we are good, AFAICS!

Reply:
oh thanks for looking into that! Yes, a bit misleading.